src/containers/AppLoader/index.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { View, ActivityIndicator, StyleSheet } from 'react-native';
+import { object } from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import useSessionChangeEffect from 'hooks/useSessionChangeEffect';
+import { LOGIN_SCREEN, MAIN_SCREEN } from 'constants/screens';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: '#FAFAFA'
+  }
+});
+
+export default function AppLoader({ navigation }) {
+  useSessionChangeEffect(
+    user => {
+      const userLoggedIn = !isEmpty(user);
+      if (userLoggedIn) {
+        navigation.navigate(MAIN_SCREEN);
+      } else {
+        navigation.navigate(LOGIN_SCREEN);
+      }
+    },
+    [navigation]
+  );
+
+  return (
+    <View style={styles.container}>
```

Comment:
Is this going to show a loading indicator when is checking the session?

I think we should show the splash screen until the app is ready to launch. Otherwise, this screen will look like a flickering screen (splash screen => loading screen for a second => user screen). WDYT?

Reply:
Yes, I think that would look better, we would need to come up with some kind of "fake" splash though, since RNNav needs to display something and does not have control over the current splash.

Reply:
Isn't possible to use `react-native-splash-screen` to programmatically hide the splash screen?

Reply:
https://github.com/crazycodeboy/react-native-splash-screen this one? Looks like we could, my only concern would be RN 0.60 support, last update was on February 🤔